### PR TITLE
Add MimicryNFTMarketIndex spec

### PR DESCRIPTION
**Pull request by AslanGoldenhour**
👋 Hi all!

I've added the 2nd data spec needed for Mimicry. Please review and let me know if you have any questions. 

As an FYI, I've included information in the spec for requesting the needed data on Eth Mainnet via NFTGo, and on Solana via Banksea Finance. I recognize the centralization risk of only starting with one data source for each chain, and think it's okay until we validate the demand of a Mimicry Market using this feed.

**To start we'll plan fund feeds for:**
```s
# the market capitalization of the 50 most valuable NFT collections on the Ethereum Mainnet, represented in USD
queryData = abi.encode("MimicryNFTMarketIndex", abi.encode(0, 0))
```

and 

```s
# the market capitalization of the 50 most valuable NFT collections on the Solana Mainnet, represented in SOL
queryData = abi.encode("MimicryNFTMarketIndex", abi.encode(1, 2))
```

p.s. all things being equal, the Eth Mainnet feed is definitely the more important of the two.

### CODE REVIEW

File: types/MimicryNFTMarketIndex.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+The `MimicryNFTMarketIndex` query has two parameters:
+
+1. **chainId**
+    - description: The chain id of the blockchain that the collections live on
+    - value type: `uint256`
+    - allowed values:
+        - `0` - ETH Mainnet
+        - `1` - Solana Mainnet
+2. **currency**
+    - description: The currency used to calculate the market capitalization
```
Comment (themandalore) [inline types/MimicryNFTMarketIndex.md:39]: I think we can just use string here.  Most reporters are familiar w/lower case currency symbols for the spot price

Comment (AslanGoldenhour) [reply types/MimicryNFTMarketIndex.md]: Thanks for the review. I changed the currency query param to a string and added a note about the impact of this in the dispute section. 

Generally speaking I assume that any reporting using Banksea for Solana will report in SOL, and any reporter using NFTGo for Eth Mainnet will use either USD or ETH. 

And then reporters who build transaction history from scratch via Reservoir (or similar) can better handle reporting in whatever currency is spec'd.

File: types/MimicryNFTMarketIndex.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+## Query Parameters
+
+The `MimicryNFTMarketIndex` query has two parameters:
+
+1. **chainId**
+    - description: The chain id of the blockchain that the collections live on
+    - value type: `uint256`
+    - allowed values:
+        - `0` - ETH Mainnet
+        - `1` - Solana Mainnet
```
Comment (themandalore) [inline types/MimicryNFTMarketIndex.md:37]: I would also make chain a string, or if you're only using EVM's , use a number.  The issue is that chainID is an actual variable in EVM blockchains where 1 is ETH Mainnet, so this is super confusing.

Comment (AslanGoldenhour) [reply types/MimicryNFTMarketIndex.md]: Roger that. Just made the change. The thinking on my end with IDs was that reporters will probably code up a switch statement (or something similar) and only support a small number of chains to start. Easy enough for them to switch against a string though.

**Comment by AslanGoldenhour** [comment]
@themandalore I think this is all set now. Can you please re-review and let me know if anything else is needed before merging this PR?

**Comment by themandalore** [comment]
> @themandalore I think this is all set now. Can you please re-review and let me know if anything else is needed before merging this PR?

Looks good, I think Akrem is going to review/ make some code in telliot and will then merge it!

**Review by themandalore — APPROVED** [review]

**Pull request merged**